hw08-json/src/main/java/otus/alexmv/hw08/writer/JsonObjectWriter.java

```diff
@@ -0,0 +1,79 @@
+package otus.alexmv.hw08.writer;
+
+import javax.json.*;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.Objects;
+
+import static otus.alexmv.hw08.writer.JsonObjectHelper.resolveType;
+
+public class JsonObjectWriter {
+
+    public JsonValue buildJsonValue(Object object) {
+        if (Objects.isNull(object))
+            return JsonValue.NULL;
+
+        switch (resolveType(object)) {
+            case PRIMITIVE:
+                return getJsonPrimitive(object);
+            case OBJECT:
+                return getJsonObject(object);
+            case ARRAY:
+                return getJsonArray(object);
+            case COLLECTION:
+                return getJsonArray(((Collection<?>) object).toArray());
+            default:
+                break;
+        }
+        throw new RuntimeException("default");
```

Comment:
```suggestion
        throw new RuntimeException("Unknown object type " + object.getClass().getCanonicalName());
```

Так будет понятнее, в чем именно была проблема.